sdk/go/common/workspace/plugins.go

```diff
@@ -1570,8 +1571,7 @@ func getPluginInfoAndPath(
 
 	// If we have a version of the plugin on its $PATH, use it, unless we have opted out of this behavior explicitly.
 	// This supports development scenarios.
-	optOut, isFound := os.LookupEnv("PULUMI_IGNORE_AMBIENT_PLUGINS")
-	includeAmbient := !(isFound && cmdutil.IsTruthy(optOut)) || isBundled
+	includeAmbient := !(env.IgnoreAmbientPlugins.Value()) || isBundled
```

Comment:
This is a good example of how the env var library code will be cleaner than what we currently have.